tools/hardware/config.py

```diff
@@ -50,6 +72,20 @@ def get_bootloader_reserve(self) -> int:
         of physical memory. Mark it as unavailable. '''
         return self.MEGA_PAGE_SIZE
 
+    def align_memory(self, regions: Set[Region]) -> List[Region]:
+        ''' RISC-V expects physBase to be the address that the bootloader is loaded at. '''
```

Comment:
I have some doubt that this is really what RISC-V wants. This just seem the be something that is currently there due to the history of the RISC-V port - and it feels wrong that we make assumption about the bootloader. If you plan to merge this PR now, it is at least helpful bring some light on the current implementation. But the comment could say this is not the intended long term solution because this creates a nasty dependency on the botloader and how big it is. The other PR (https://github.com/seL4/seL4/pull/365) I'm working at will remove all this code again and use DTB overlays to reserve bootloader memory and align things with how it's done on ARM.

Reply:
Happy to add that